specs/data-grid/data-grid.spec.md

```diff
@@ -0,0 +1,302 @@
+- `fast-data-grid-header`
+
+*Attributes:*
+- none
+
+*properties:*
+- `columnsData`  
+An array of `DataGridColumn` objects that define what columns will be displayed in the grid.  The order of the columns determines their order in the grid.
+
+- `headerCellItemTemplate`  
+Custom [template](https://fast.design/docs/fast-element/declaring-templates) to use when generating header cells by iterating over data.  The default template uses `fast-data-grid-header-cell`, this is where authors change that.
+
+*Slots:*
+- `default`  
+Default slot for items
+
+*Events*
+- none
+
+*parts:*
+- `headerCellsSlot`
+
+
+**Data grid header cell**
+- `fast-data-grid-header-cell`
+
+*Attributes:*
+- extends HTML Element attributes
+
+*properties:*
+- `columnData`  
+The `DataGridColumn` this column header represents.
+
+*Slots:*
+- `default`  
+Default slot for items
+
+*Events*
+- none
+
+*parts:*
+- `headerCellsSlot`
+
+
+**Data grid row**
+- `fast-data-grid-row`
+
+*Attributes:*
+- `gridTemplateColumns`  
+String that gets applied to the the css gridTemplateColumns attribute for the row
+
+*properties:*
+- `rowData`  
+The objects that contains the data to be displayed in this row.
+
+- `columnsData`  
+An array of `DataGridColumn` objects that define what columns will be displayed in the grid.  The order of the columns determines their order in the grid.
+
+- `cellItemTemplate`  
+Custom [template](https://fast.design/docs/fast-element/declaring-templates) to use when generating cells by iterating over data.  The default template uses `fast-data-grid-cell`, this is where authors change that.
+
+*Slots:*
+- `default`  
+Default slot for items
+
+*Events*
+- none
+
+*parts:*
+- `cellsSlot`
+
+**Data grid cell**
+*Component name:*
+- `fast-data-grid-cell`
+
+*Attributes:*
+- `gridColumnIndex`  
+The grid column this cell appears in.
+
+*properties:*
+- `rowData`  
+The object that contains the data to be displayed in this row.
+
+- `columnData`  
+The `DataGridColumn` this cell represents.
+
+*Slots:*
+- `default`  
+Default slot for items
+
+*Events*
+- none
+
+*parts:*
+- `cellSlot`
+
+## Implementation
+
+The most basic use case for this component would be to just pass it an array of objects for it to render.  
+
+For example given a grid component:
+```html
+ <fast-data-grid id="defaultGrid"></fast-data-grid>
+ ```
+
+ And some data:
+ ```js
+const baseRows: object[] = [
+    { name: "Rob", age: "19" },
+    { name: "Bob", age: "20" },
+];
+ ```
+
+ An author could pass the data to the component from a javascript function:
+
+```js
+onLoad(): void {
+   const defaultGrid: DataGrid | null = document.getElementById(
+        "defaultGrid"
+    ) as DataGrid;
+    if (defaultGrid !== null) {
+        defaultGrid.rowsData = baseRows;
+    }
+}
+```
+
+This renders a basic grid with a column titled "name" and another titled "age".  In addition to the title row there would be two rows of data populated with the values for name and age.  Note that data
```

Comment:
"Note that data" <= not sure what is meant to be noted here.

Reply:
Me either.  Deleted.